### PR TITLE
Support cross-account access

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR changes this module from creating a user that must be the same account as the SSM Parameter Store and the AMI to supporting the case where the user is created in a different account, but still has the necessary permissions.

## 💭 Motivation and Context
As we move to the COOL, we need to update this module so that it supports creating a user in a different account than where the SSM Parameter Store and AMIs are located. 

## 🧪 Testing

Testing was done by executing `terraform apply` and `terraform delete` with a variety of inputs in the `examples/read_a_single_parameter` code and manually verifying that the resulting user had all of the necessary permissions to read SSM parameter store parameters and create AMIs.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,10 +23,16 @@ details on Terraform modules and the standard module structure.
...
+    aws.images-ProvisionParameterStoreReadRoles = aws.images-ProvisionParameterStoreReadRoles
+    aws.users                                   = aws.users
+  }
+
+  images_account_id = "111111111111"
+  ssm_parameters    = ["/github/oauth_token"]
+  user_name         = "test-ansible-role-cyhy-core"
   tags = {
-    Team        = "NCATS OIS - Development"
+    Team        = "VM Fusion - Development"
```
Comment (felddy) [inline README.md:35]: We can't update this fast enough... not-lol

**Review by felddy — APPROVED** [review]
Looks good.  Thanks for conquering this! ⛑

**Review by mcdonnnj — APPROVED** [review]
Lookin' good. Nice work 👍

File: README.md [hunk 2]
```diff
@@ -31,10 +46,14 @@ module "iam_user" {
...
-| user_name | The name to associate with the AWS IAM user (e.g. test-ansible-role-cyhy-core) | string | | yes |
 | add_packer_permissions | Whether or not to give the IAM user the permissions needed by packer to create an AMI | bool | `false` | no |
+| ec2amicreate_role_name | The name of the IAM role in the Images account that allows all of the EC2 actions needed to create an AMI. | string | `EC2AMICreate` | no |
+| images_account_id | The AWS account ID containing the SSM parameter store that the IAM user needs to read from and also where the user will be allowed to create images (if add_packer_permissions is true); if not provided, the current calling account ID is used | string | ID of calling account | no |
+| parameterstorereadonly_role_description | The description to associate with the IAM role (as well as the corresponding policy) that allows read-only access to the specified SSM Parameter Store parameters.  Note that a \"%s\" in this value will get replaced with the user_name variable. | string | `Allows read-only access to SSM Parameter Store parameters required for %s.` | no |
```
Comment (jsf9k) [inline README.md]: You don't need the backslashes here.  Only in `variables.tf`.

File: README.md [hunk 3]
```diff
@@ -31,10 +46,14 @@ module "iam_user" {
...
+| parameterstorereadonly_role_name | The name to assign the IAM role (as well as the corresponding policy) that allows read-only access to the specified SSM Parameter Store parameters.  Note that a \"%s\" in this value will get replaced with the user_name variable. | string | `ParameterStoreReadOnly-%s` | no |
```
Comment (jsf9k) [inline README.md]: Backslashes again.

File: examples/read_a_single_parameter/main.tf [hunk 4]
```diff
@@ -1,16 +1,36 @@
-# Configure AWS
+# Default AWS provider (required to set region, but not used)
```
Comment (jsf9k) [inline examples/read_a_single_parameter/main.tf]: I didn't realize that you would have to define the default provider even if you don't use it.  This makes me think that we should make one of the other providers the default.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I have some minor change requests/discussion points.

File: README.md (hunk 1 above)
Comment (dav3r) [reply README.md:35]: For realz.  ♻️

File: README.md (hunk 2 above)
Comment (dav3r) [reply README.md]: Great catch - fixed in https://github.com/cisagov/molecule-packer-ci-iam-user-tf-module/pull/8/commits/11770729a3a239e6eb33b58997a80e81abbec419

File: README.md (hunk 3 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/molecule-packer-ci-iam-user-tf-module/pull/8/commits/11770729a3a239e6eb33b58997a80e81abbec419

File: examples/read_a_single_parameter/main.tf (hunk 4 above)
Comment (dav3r) [reply examples/read_a_single_parameter/main.tf]: Thanks for mentioning this.  I had thought about doing exactly that, but needed another voice to push me over the edge.  This has now been changed in https://github.com/cisagov/molecule-packer-ci-iam-user-tf-module/pull/8/commits/9b3eb1d2f076d02e2dcf168cb7788ab0f902c028.

**Review by jsf9k — APPROVED** [review]
Powerful work!

**Review by felddy — APPROVED** [review]
Things can only get better!
🎼 [Wow wow wow oh, wow wow wow oh oh oh oh](https://www.youtube.com/watch?v=-OO9LloDSJo) 🎶

**Pull request merged**